src/parquet/util/buffer.h

```diff
@@ -127,12 +129,39 @@ class ResizableBuffer : public MutableBuffer {
 // garbage-collected
 class OwnedMutableBuffer : public ResizableBuffer {
  public:
-  OwnedMutableBuffer();
-  virtual void Resize(int64_t new_size);
+  explicit OwnedMutableBuffer(int64_t size = 0, MemoryPool* pool = get_default_pool());
+  virtual ~OwnedMutableBuffer();
+  void Resize(int64_t new_size) override;
+  void Reserve(int64_t new_capacity);
+  uint8_t& operator[](int64_t i);
 
  private:
   // TODO: aligned allocations
-  std::vector<uint8_t> buffer_owner_;
+  MemoryPool* memory_pool_;
+
+  OwnedMutableBuffer(OwnedMutableBuffer& buffer) = delete;
+  OwnedMutableBuffer& operator=(OwnedMutableBuffer& buffer) = delete;
+};
+
+template <class T>
+class DataBuffer {
```

Comment:
Let's call this something semantically distinct, like `Vector` -- my objection to the prior layering was about object semantics.